trie/trieStorageManager.go

```diff
@@ -480,18 +522,24 @@ func (tsm *trieStorageManager) takeSnapshot(snapshotEntry *snapshotsQueueEntry,
 	log.Trace("trie snapshot started", "rootHash", snapshotEntry.rootHash)
 
 	newRoot, err := newSnapshotNode(tsm, msh, hsh, snapshotEntry.rootHash)
-	if err == ErrContextClosing {
+	if isClosingError(err) {
 		log.Debug("context closing when creating a new snapshot node")
 		return
 	}
 	if err != nil {
-		log.Error("takeSnapshot: trie storage manager: newSnapshotTrie", "rootHash", snapshotEntry.rootHash, "error", err.Error())
+		log.Error("takeSnapshot: trie storage manager: newSnapshotTrie",
+			"rootHash", snapshotEntry.rootHash,
+			"main trie rootHash", snapshotEntry.mainTrieRootHash,
+			"error", err.Error())
 		return
 	}
 
-	stsm, err := newSnapshotTrieStorageManager(tsm)
+	stsm, err := newSnapshotTrieStorageManager(tsm, snapshotEntry.epoch)
 	if err != nil {
```

Comment:
did you forgot to check the error if is of type closing error?
I would change all these "duplicated" error management blocks. Something like:
```GO
    err := do_something_here
    if err != nil{
        treatError(err, "custom message", <roothash>)
        return
    }

//.....
func treatError(err error, message string, rootHash []byte){
    if isClosingError(err){
        log.Debug("context closing", "message", message)
        return
    }

    log.Error(message, "rootHash", rootHash, "err", err.Error())
}

Reply:
There is no need to check for closingError here, because this func does not access the db. 
Refactored as suggested.